frontend/projects/upgrade/src/app/features/dashboard/segments/components/segment-members/segment-members.component.ts

```diff
@@ -164,12 +164,13 @@ export class SegmentMembersComponent implements OnInit, OnChanges {
           ) {
             this.subSegmentIds.push(segment.name);
             this.segmentNameId.set(segment.name, segment.id);
-          }
-        } else {
-          if (segment.type !== SEGMENT_TYPE.GLOBAL_EXCLUDE && segment.context === this.currentContext) {
+          } else if (segment.type !== SEGMENT_TYPE.GLOBAL_EXCLUDE && segment.id !== this.segmentInfo.id && this.currentContext==="ALL") {
```

Comment:
Fix the lint in the line and please refactor the code there are too many if statements.
Use array map and filters.